cmd/gctcli/data_history.go

```diff
@@ -139,34 +63,228 @@ var dataHistoryCommands = &cli.Command{
 			ArgsUsage: "<nickname>",
 			Action:    getDataHistoryJobSummary,
 			Flags: []cli.Flag{
-				&cli.StringFlag{
-					Name:  "nickname",
-					Usage: "binance-spot-btc-usdt-2019-trades",
-				},
+				nicknameFlag,
 			},
 		},
+		dataHistoryJobCommands,
 		{
-			Name:   "addnewjob",
-			Usage:  "creates a new data history job",
-			Flags:  fullJobSubCommands,
-			Action: upsertDataHistoryJob,
+			Name:      "deletejob",
+			Usage:     "sets a jobs status to deleted so it no longer is processed",
+			ArgsUsage: "<id> or <nickname>",
+			Flags:     specificJobSubCommands,
+			Action:    setDataHistoryJobStatus,
 		},
 		{
-			Name:   "upsertjob",
-			Usage:  "adds a new job, or updates an existing one if it matches jobid OR nickname",
-			Flags:  fullJobSubCommands,
-			Action: upsertDataHistoryJob,
+			Name:      "pausejob",
+			Usage:     "sets a jobs status to paused so it no longer is processed",
+			ArgsUsage: "<id> or <nickname>",
+			Flags:     specificJobSubCommands,
+			Action:    setDataHistoryJobStatus,
 		},
 		{
-			Name:      "deletejob",
-			Usage:     "sets a jobs status to deleted so it no longer is processed",
+			Name:      "unpausejob",
+			Usage:     "sets a jobs status to active so it can be processed",
 			ArgsUsage: "<id> or <nickname>",
 			Flags:     specificJobSubCommands,
-			Action:    deleteDataHistoryJob,
+			Action:    setDataHistoryJobStatus,
+		},
+		{
+			Name:      "updateprerequisite",
+			Usage:     "adds or updates a prerequisite job to the job referenced - if the job is active, it will be set as 'paused'",
+			ArgsUsage: "<prerequisite> <nickname>",
+			Flags:     prerequisiteJobSubCommands,
+			Action:    setPrerequisiteJob,
+		},
+		{
+			Name:      "removeprerequisite",
+			Usage:     "removes a prerequisite job from the job referenced - if the job is 'paused', it will be set as 'active'",
+			ArgsUsage: "<nickname>",
+			Flags: []cli.Flag{
+				nicknameFlag,
+			},
+			Action: setPrerequisiteJob,
+		},
+	},
+}
+
+var dataHistoryJobCommands = &cli.Command{
+	Name:      "addjob",
+	Usage:     "add or update data history jobs",
+	ArgsUsage: "<command> <args>",
+	Subcommands: []*cli.Command{
+		{
+			Name:   "savecandles",
+			Usage:  "will fetch candle data from an exchange and save it to the database",
+			Flags:  append(baseJobSubCommands, dataHandlingJobSubCommands...),
+			Action: upsertDataHistoryJob,
+		},
+		{
+			Name:   "convertcandles",
+			Usage:  "convert candles saved to the database to a new resolution eg 1min -> 5min",
+			Flags:  append(baseJobSubCommands, candleConvertJobJobSubCommands...),
+			Action: upsertDataHistoryJob,
+		},
+		{
+			Name:   "savetrades",
```

Comment:
```json
--request_size_limit value  the number of candles to retrieve per API request (default: 500)
```

Might be better to change to trades?

Also im confused by this debug output. Does this mean only 5 trades are loaded?
```json
job ftxtest4 request size 500 outside limit of 5, defaulting to 5
```

Reply:
Its a bit more complicated, but highlighted here: https://github.com/thrasher-corp/gocryptotrader/blob/master/engine/datahistory_manager.md#candle-intervals-and-trade-fetching 
5 intervals worth of trades will be fetched. But I'll rethink things

Reply:
Thanks for clarifying!